### PR TITLE
Update auth methods

### PR DESCRIPTION
**Pull request by mediumTaj**
### Summary
This pull request is to the 4.0.0 base branch for the major release. It updates the generated services to use `Authenticator` rather than `AuthConfig` to authenticate services. Legacy constructors were also removed from the generated service. All tests and examples were also updated. This branch depends on a .NET SDK core pre-release `1.0.0-rc1` from Nuget.

### CODE REVIEW

File: examples/IBM.Watson.Assistant.v2.Examples/IBM.Watson.Assistant.v2.Examples.csproj [hunk 1]
```diff
@@ -6,10 +6,11 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="IBM.Cloud.SDK.Core" Version="0.8.0" />
+    <PackageReference Include="IBM.Cloud.SDK.Core" Version="1.0.0-rc1" />
   </ItemGroup>
 
   <ItemGroup>
+    <!-- <ProjectReference Include="..\..\..\dotnet-sdk-core\src\IBM.Cloud.SDK.Core\IBM.Cloud.SDK.Core.csproj" /> -->
```
Comment (lpatino10) [inline examples/IBM.Watson.Assistant.v2.Examples/IBM.Watson.Assistant.v2.Examples.csproj]: Just for verification, is this just commented out until the core changes merge to a public release?

Comment (mediumTaj) [reply examples/IBM.Watson.Assistant.v2.Examples/IBM.Watson.Assistant.v2.Examples.csproj]: Yep, these will be here until the major release so I can reference the nuget package vs the local core reference!

**Comment by mediumTaj** [comment]
@lpatino10 Updated one more time if you can take a look

File: src/IBM.Watson.CompareComply.v1/Model/KeyValuePair.cs [hunk 2]
```diff
@@ -31,10 +30,10 @@ public class KeyValuePair
         [JsonProperty("key", NullValueHandling = NullValueHandling.Ignore)]
         public Key Key { get; set; }
         /// <summary>
-        /// A list of values in a key-value pair.
+        /// A value in a key-value pair.
         /// </summary>
         [JsonProperty("value", NullValueHandling = NullValueHandling.Ignore)]
-        public List<Value> Value { get; set; }
```
Comment (lpatino10) [inline src/IBM.Watson.CompareComply.v1/Model/KeyValuePair.cs]: Looks like this got reverted

Comment (mediumTaj) [reply src/IBM.Watson.CompareComply.v1/Model/KeyValuePair.cs]: Yeah, I generated using the last commit we generated from to keep the changes minimal. I forgot this was a hand-edit!

**Comment by lpatino10** [comment]
Unfortunately still seeing some issues 😕

**Comment by mediumTaj** [comment]
@lpatino10 got it - was missing some null checks in the core

**Pull request merged**